src/main/java/net/pms/util/ImdbUtil.java

```diff
@@ -1,43 +1,223 @@
+						ArrayList<Entry<Path, Double>> candidatesList = new ArrayList<>(candidates.entrySet());
+						if (candidatesList.size() > 1) {
+							// Sort by score
+							Collections.sort(candidatesList, new Comparator<Entry<Path, Double>>() {
+								@Override
+								public int compare(Entry<Path, Double> o1, Entry<Path, Double> o2) {
+									return o2.getValue().compareTo(o1.getValue());
+								}
+							});
+						}
+						nfoFile = candidatesList.get(0).getKey();
+					}
+				} catch (IOException e) {
+					LOGGER.error(
+						"An error occured when trying to scan for nfo files belonging to \"{}\": {}",
+						file,
+						e.getMessage()
+					);
+					LOGGER.trace("", e);
+				}
+			}
+		}
+		if (!Files.exists(nfoFile)) {
+			LOGGER.debug("Didn't find a matching nfo file for \"{}\" to search for IMDB ID", file);
+			return null;
+		}
+		try {
+			if (Files.size(nfoFile) > 100000) {
+				LOGGER.debug(
+					"Skipping search for IMDB ID in \"{}\" since it's too large ({})",
+					nfoFile,
+					StringUtil.formatBytes(Files.size(nfoFile), true)
+				);
+				return null;
+			}
+		} catch (IOException e) {
+			LOGGER.warn("Failed to get size of \"{}\", skipping search for IMDB ID: {}", nfoFile, e.getMessage());
+			LOGGER.trace("", e);
+			return null;
+		}
+
+		Charset charset = null;
+		try {
+			charset = FileUtil.detectCharset(nfoFile, 70);
+		} catch (IOException e) {
+			LOGGER.error(
+				"An error occurred while trying to detect the character encoding of \"{}\": {}",
+				nfoFile,
+				e.getMessage()
+			);
+			LOGGER.trace("", e);
+		}
+		if (charset == null) {
+			if (Platform.isWindows()) {
+				// Because UMS is configured to erroneously always set the default charset to UTF-8, this is useless for Windows
+				charset = WinUtils.getOEMCharset();
+				if (charset == null) {
+					charset = StandardCharsets.US_ASCII;
+				}
+			} else {
+				charset = Charset.defaultCharset();
+			}
+			LOGGER.debug(
+				"Failed to detect the character encoding of \"{}\", falling back to the default charset \"{}\"",
+				nfoFile,
+				charset
+			);
+		}
+		LOGGER.trace("Scanning \"{}\" for IMDB ID for \"{}\"", nfoFile, file);
+		try (BufferedReader reader = Files.newBufferedReader(nfoFile, charset)) {
+			String line = reader.readLine();
+			while (line != null) {
+				Matcher matcher = NFO_IMDB_ID.matcher(line);
+				if (matcher.find()) {
+					LOGGER.trace("Found IMDB ID {} for \"{}\" in \"{}\"", matcher.group(1), file, nfoFile);
+					return matcher.group(1);
+				}
+				line = reader.readLine();
+			}
+		} catch (IOException e) {
+			LOGGER.error("An error occurred while scanning \"{}\" for IMDB ID: {}", nfoFile, e.getMessage());
+			LOGGER.trace("", e);
+		}
+		LOGGER.trace("Failed to find IMDB ID for \"{}\" in \"{}\"", file, nfoFile);
+		return null;
 	}
 
-	public static String ensureTT(String s) {
-		return (s.startsWith("tt") ? s : "tt" + s);
+	/**
+	 * Makes sure the IMDB ID starts with "{@code tt}" by prefixing it if
```

Comment:
Sorry, i know nothing about. What is "tt" used for ?

Reply:
If you look at any IMDB link, the ID starts with ```tt```. I have no idea why, you'd have to ask IMDB. As this is a fixed prefix and not really useful to keep, some simply skip it. These are methods for stripping or adding the ```tt``` prefix according to what is expected where you are going to use it.